parse/lex.go

```diff
@@ -279,9 +296,13 @@ func lexPattern(l *lexer) stateFn {
 			}
 
 			return lexExpr(l)
-		case !l.isComplexMessage && len(s) == 0 && r == '.':
+		case !l.isComplexMessage && (len(s) == 0 || isStringWhitespace(s)) && r == '.':
```

Comment:
`isStringWhitespace` is called twice - once here and second time in `mkText`. Should it be like this?

```suggestion
		case !l.isComplexMessage && isStringWhitespace(s) && r == '.':
			l.backup()
			return l.emitItem(mk(itemWhitespace, s))
		case !l.isComplexMessage && len(s) == 0 && r == '.':
```